src/mod_roster.erl

```diff
@@ -712,11 +691,13 @@ process_subscription(Direction, User, Server, JID1, Type, Reason) ->
 autoreply_to_type(subscribed) -> <<"subscribed">>;
 autoreply_to_type(unsubscribed) -> <<"unsubscribed">>.
 
-process_subscription_transaction(Direction, LUser, LServer, LJID, Type, Reason) ->
-    Item = case mod_roster_backend:get_roster_entry_t(LUser, LServer, LJID, full) of
+process_subscription_transaction(Direction, JID, LJID, Type, Reason) ->
+    #jid{luser = LUser, lserver = LServer} = JID,
+    Item = case get_roster_entry_t(JID, LJID, full) of
```

Comment:
I think we do not need the 'lowered' JID in the argument, I would still pass the original #jid record here and do `jid = jid:to_lower(JID)` below. I would also call it something else than just JID/LJID to differentiate from the first argument.